test/Microsoft.Azure.Devices.Edge.Test.Common/PartitionReceiveHandler.cs

```diff
@@ -1,39 +0,0 @@
-// Copyright (c) Microsoft. All rights reserved.
-
-namespace Microsoft.Azure.Devices.Edge.Test.Common
-{
-    using System;
-    using System.Collections.Generic;
-    using System.Threading.Tasks;
-    using Microsoft.Azure.EventHubs;
-
-    class PartitionReceiveHandler : IPartitionReceiveHandler
-    {
-        readonly Func<EventData, bool> onEventReceived;
-
-        public PartitionReceiveHandler(Func<EventData, bool> onEventReceived)
-        {
-            this.onEventReceived = onEventReceived;
-        }
-
-        public int MaxBatchSize { get; set; }
-
-        public Task ProcessEventsAsync(IEnumerable<EventData> events)
-        {
-            if (events != null)
```

Comment:
This null check doesn't exist in the new logic. Assuming the new api for `client.ReadEventsFromPartitionAsync()` won't return null, but an empty enumerable?

Reply:
Good catch. I believe the new code will never get a null `PartitionEvent` due to the way async enumerables work (`ReadEventsFromPartitionAsync()` can return null, but `async foreach` handles it). However, I found some sample code in the Azure.Messaging.EventHubs repo that suggests the `EventData` inside a `PartitionEvent` [_can_ be null](https://github.com/Azure/azure-sdk-for-net/blob/e09d8fbab8f9966a2303104b037e56c31d7bbcf1/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample05_ReadingEvents.md#L258). So I'll add a check to the new code.

Reply:
Fixed.